In [1]:
import pandas as pd
import numpy as np 
import matplotlib.pyplot as plt 
import seaborn as sns

In [3]:
continents = pd.read_csv('../data/continents.csv')
gdp_df = pd.read_csv('../data/gdp_per_capita.csv')

In [31]:
gdp_df.head(10)

Country  Year  GDP_Per_Capita
0  Afghanistan  2021     1517.016266
1  Afghanistan  2020     1968.341002
2  Afghanistan  2019     2079.921861
3  Afghanistan  2018     2060.698973
4  Afghanistan  2017     2096.093111
5  Afghanistan  2016     2101.422187
6  Afghanistan  2015     2108.714173
7  Afghanistan  2014     2144.449634
8  Afghanistan  2013     2165.340915
9  Afghanistan  2012     2122.830759

In [12]:
gdp_df.tail(10)

Country or Area  Year        Value  Value Footnotes
7652        Zimbabwe  1999  2866.032886              NaN
7653        Zimbabwe  1998  2931.725144              NaN
7654        Zimbabwe  1997  2896.147308              NaN
7655        Zimbabwe  1996  2867.026043              NaN
7656        Zimbabwe  1995  2641.378271              NaN
7657        Zimbabwe  1994  2670.106615              NaN
7658        Zimbabwe  1993  2458.783255              NaN
7659        Zimbabwe  1992  2468.278257              NaN
7660        Zimbabwe  1991  2781.787843              NaN
7661        Zimbabwe  1990  2704.757299              NaN

In [17]:
gdp_df = gdp_df.drop(columns = 'Value Footnotes')

In [23]:
gdp_df = gdp_df.rename(columns={'Country or Area' : 'Country', 'Value' : 'GDP_Per_Capita'})

In [50]:
gdp_df.head()
# checking to see if dropping and renaming worked

Country  Year  GDP_Per_Capita
0  Afghanistan  2021     1517.016266
1  Afghanistan  2020     1968.341002
2  Afghanistan  2019     2079.921861
3  Afghanistan  2018     2060.698973
4  Afghanistan  2017     2096.093111

In [54]:
gdp_df.shape
# trying different ways to get columns and rows
# this gives rows, columns

(7662, 3)

In [56]:
gdp_df.shape[0]
# gives rows

7662

In [57]:
gdp_df.shape[1]
# gives columns

3

In [73]:
gdp_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7662 entries, 0 to 7661
Data columns (total 3 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country         7662 non-null   object 
 1   Year            7662 non-null   int64  
 2   GDP_Per_Capita  7662 non-null   float64
dtypes: float64(1), int64(1), object(1)
memory usage: 179.7+ KB


In [61]:
gdp_df.count()
# number of values? in each columns?

Country           7662
Year              7662
GDP_Per_Capita    7662
dtype: int64

In [63]:
len(gdp_df.index)
# number of rows ?

7662

In [48]:
len(gdp_df.axes[0])
# gives the rows

7662

In [51]:
# gives the columns 
len(gdp_df.axes[1])

3

In [52]:
len(gdp_df)
# rows

7662

In [58]:
len(gdp_df.columns)
# columns

3

In [74]:
gdp_df.size
# not what I am looking for

22986

In [76]:
gdp_df.dtypes

Country            object
Year                int64
GDP_Per_Capita    float64
dtype: object

Question: How many rows and columns does gdp_df have? What are the data types of its columns? If any of the columns are not the expected types, figure out why and fix it.
- rows: 7662
- columns: 3 (not including the index(not sure if that is counted))
- data types: country - object, year - int64, gdp_per_capita - float64
- changed county data type to string. 
t64